rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/indices.create/20_synthetic_source.yml

```diff
@@ -1553,3 +1552,402 @@ synthetic_source with copy_to and invalid values for copy:
+        body:
+          mappings:
+            _source:
+              mode: synthetic
+            properties:
+              name:
+                type: keyword
+              my_values:
+                properties:
+                  k:
+                    type: keyword
+                    ignore_above: 1
+                    copy_to: c.copy
+                  long:
+                    type: long
+                    copy_to: c.copy
+              c:
+                properties:
+                  copy:
+                    type: keyword
+
+  - do:
+      index:
+        index: test
+        id: 1
+        refresh: true
+        body:
+          name: "A"
+          my_values:
+            k: "hello"
+            long: 100
+
+  - do:
+      index:
+        index: test
+        id: 2
+        refresh: true
+        body:
+          name: "B"
+          my_values:
+            k: ["55", "66"]
+            long: [77, 88]
+
+  - do:
+      index:
+        index: test
+        id: 3
+        refresh: true
+        body:
+          name: "C"
+          my_values:
+            k: "hello"
+            long: 100
+          c:
+            copy: "zap"
+
+  - do:
+      search:
+        index: test
+        sort: name
+        body:
+          docvalue_fields: [ "c.copy" ]
+
+  - match:
+      hits.hits.0._source:
+        name: "A"
+        my_values:
+          k: "hello"
+          long: 100
+  - match:
+      hits.hits.0.fields:
+        c.copy: [ "100", "hello" ]
+
+  - match:
+      hits.hits.1._source:
+        name: "B"
+        my_values:
+          k: ["55", "66"]
+          long: [77, 88]
+  - match:
+      hits.hits.1.fields:
+        c.copy: ["55", "66", "77", "88"]
+
+  - match:
+      hits.hits.2._source:
+        name: "C"
+        my_values:
+          k: "hello"
+          long: 100
+        c:
+          copy: "zap"
+  - match:
+      hits.hits.2.fields:
+          c.copy: [ "100", "hello", "zap" ]
+
+---
+synthetic_source with copy_to pointing to ambiguous field:
+  - requires:
+      cluster_features: ["mapper.source.synthetic_source_copy_to_inside_objects_fix"]
+      reason: requires copy_to support in synthetic source
+
+  - do:
+      indices.create:
+        index: test
+        body:
+          mappings:
+            _source:
+              mode: synthetic
+            properties:
+              k:
+                type: keyword
+                copy_to: a.b.c
+              a:
+                properties:
+                  b:
+                    properties:
+                      c:
+                        type: keyword
+                  b.c:
+                    type: keyword
+
+  - do:
+      index:
+        index: test
+        id: 1
+        refresh: true
+        body:
+          k: "hey"
+
+  - do:
+      search:
+        index: test
+        body:
+          docvalue_fields: [ "a.b.c" ]
+
+  - match:
+      hits.hits.0._source:
+        k: "hey"
+  - match:
+      hits.hits.0.fields:
+        a.b.c: [ "hey" ]
```

Comment:
Why do we decide to match `a.b.c` versus `a.b\.c'? I was wondering about that in `findParentObject`..

Reply:
I think I got it, we first check for a leaf field match before checking for object matches.